### ISSUE TITLE
"truffle create contract ..." generates deprecated code

### ISSUE THREAD

**Issue by gerchicov-bp** [trufflesuite/truffle#1210]
## Issue

The problem with deprecated code is described for example here:
https://ethereum.stackexchange.com/questions/47175/defining-constructors-as-functions-with-the-same-name-as-the-contract-is-depreca

## Steps to Reproduce

run "truffle create contract <some_name>"

## Expected Behavior

generated code:
```
pragma solidity ^0.4.24;


contract <some_name> {
  constructor() public {
    // constructor
  }
}
```

## Actual Results

generated code:
```
pragma solidity ^0.4.4;


contract Mortal {
  function Mortal() {
    // constructor
  }
}
```

## Environment

* Truffle version (`truffle version`): Truffle v4.1.11 (core: 4.1.11)
* Solidity version: Solidity v0.4.24 (solc-js)
* Operating System: OSX

Other environment params don't have (and shouldn't have) influence on this bug

**Comment by cgewecke** [comment]
@gerchicov-bp This is fixed in the latest version of Truffle 4 (4.1.14). The solidity version is `^0.4.22` but the constructor syntax is correct.

**Comment by gerchicov-bp** [comment]
@cgewecke updated but syntax checker still says `No visibility specified. Defaulting to "public"`

what about to fix that?

**Comment by cgewecke** [comment]
Oops yes you're right.

**Comment by subramanianv** [comment]
Is anybody working on this issue ? I would like to take this up if its open

**Comment by cgewecke** [comment]
@subramanianv Ah actually a fix was merged yesterday in #1219 :) 

Thanks though.
